### ISSUE TITLE
Feature Request: SSH reconnect after building VM

### ISSUE THREAD

**Issue by andylawrenson** [josenk/vagrant-vmware-esxi#8]
If building an ESXi image over a relatively slow network connection (e.g. a VPN), this can take a while.

After creating the VM, createvm.rb uses the existing SSH connection to execute "vim-cmd vmsvc/getallvm" to get the VM ID - however, I have found this to fail when the VM build has taken a long time, as the existing SSH connection has been closed.

If I simply end the SSH Session and start a new one to execute the final command, then it works - which seems to be more robust.

**Comment by josenk** [comment]
BTW:  How long is "this can take a while???"  

I added keepalives to the ssh session in the createvm code.  Could you try this and let me know if it works.

# wget https://dc002.jintegrate.co/928729404826304827/vagrant-vmware-esxi-1.3.1.gem --no-check-certificate

# vagrant plugin uninstall vagrant-vmware-esxi
# vagrant plugin install ./vagrant-vmware-esxi-1.3.1.gem

**Comment by andylawrenson** [comment]
In this case, it takes about 30 mins to upload a 2Gb image.

I'd tried adding in keepalives myself (every 60 secs) without success, but have just tried your change.  This possibly seemed better, but still didn't work. 
Previously I could see "user root logged out" in the ESXI log after a few minutes of the transfer, which I saw as indicative of the the SSH session being closed - there's no timeout set in ESXi itself, and I was thinking that the SSH connection could have been broken because the network connection was so saturated from the upload that the SSH client & server were no longer able to communicate (I've seen this elsewhere over a VPN).
Trying your v1.3.1, I can no longer see the the message for "user root logged out" after a few minutes in the ESXi log, but the creation still failed with this error in the client (which is what I had before):

```
Transfer Completed
Powering on VM: at2.cnect.to
Task Completed
Completed successfully
C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/buffered_io.rb:102:in `send': closed stream (IOError)
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/buffered_io.rb:102:in `send_pending'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/connection/session.rb:271:in `block in ev_do_handle_events'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/connection/session.rb:270:in `each'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/connection/session.rb:270:in `ev_do_handle_events'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/connection/event_loop.rb:110:in `ev_select_and_postprocess'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/connection/event_loop.rb:29:in `process'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/connection/session.rb:216:in `process'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/connection/session.rb:178:in `block in loop'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/connection/session.rb:178:in `loop'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/connection/session.rb:178:in `loop'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/connection/channel.rb:269:in `wait'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh/connection/session.rb:416:in `exec!'
        from C:/Users/andy/.vagrant.d/gems/2.4.2/gems/vagrant-vmware-esxi-1.3.1/lib/vagrant-vmware-esxi/action/createvm.rb:326:in `block in createvm'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/net-ssh-4.1.0/lib/net/ssh.rb:241:in `start'
        from C:/Users/andy/.vagrant.d/gems/2.4.2/gems/vagrant-vmware-esxi-1.3.1/lib/vagrant-vmware-esxi/action/createvm.rb:67:in `createvm'
        from C:/Users/andy/.vagrant.d/gems/2.4.2/gems/vagrant-vmware-esxi-1.3.1/lib/vagrant-vmware-esxi/action/createvm.rb:16:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/Users/andy/.vagrant.d/gems/2.4.2/gems/vagrant-vmware-esxi-1.3.1/lib/vagrant-vmware-esxi/action/read_state.rb:29:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/action/builtin/handle_box.rb:56:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/Users/andy/.vagrant.d/gems/2.4.2/gems/vagrant-vmware-esxi-1.3.1/lib/vagrant-vmware-esxi/action/esxi_password.rb:18:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/action/builder.rb:116:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/action/runner.rb:66:in `block in run'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/util/busy.rb:19:in `busy'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/action/runner.rb:66:in `run'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/machine.rb:227:in `action_raw'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/machine.rb:202:in `block in action'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/environment.rb:631:in `lock'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/machine.rb:188:in `call'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/machine.rb:188:in `action'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.1/lib/vagrant/batch_action.rb:82:in `block (2 levels) in run'
```

what I'd done before that did seem to work, was simply ending the SSH session & creating a new one - here's a diff:

```
$ diff createvm.rb createvm_mod.rb
322a323,333
>           end
>
>           # reconnect
>           Net::SSH.start( config.esxi_hostname, config.esxi_username,
>             password:                   $esxi_password,
>             port:                       config.esxi_hostport,
>             keys:                       config.esxi_private_keys,
>             timeout:                    10,
>             number_of_password_prompts: 0,
>             non_interactive:            true
>           ) do |ssh|
```

many thanks, Andy

**Comment by josenk** [comment]
OK, I now close the old session and re-open a new session after the ovftool upload...   Give this one a try.

wget https://dc002.jintegrate.co/728834879475120365/vagrant-vmware-esxi-1.3.2.gem --no-check-certificate

**Comment by andylawrenson** [comment]
that worked fine now - many thanks!  Andy.

**Comment by josenk** [comment]
Thanks for the feedback...   Fixed in V1.3.2.
